samcli/commands/sync/command.py

```diff
@@ -374,31 +375,44 @@ def do_cli(
                                 dependency_layer,
                             )
                         else:
-                            execute_infra_contexts(build_context, package_context, deploy_context)
+                            infra_sync_result = execute_infra_contexts(build_context, package_context, deploy_context)
+                            code_sync_resources = infra_sync_result.code_sync_resources
+
+                            if code_sync_resources:
+                                resource_ids = [str(resource) for resource in code_sync_resources]
+
+                                LOG.info("The following resources will be code synced for an update: %s", resource_ids)
```

Comment:
Should we convert this into debug, in the case where there would be a lot of them?

Reply:
We only print this once, resource_ids is a list. We have gone through UX review with such message

Reply:
It's logical to tell customers what happens after we skip infra sync -> we queue up relevant code syncs

Reply:
My concern is, if there would be 20 items or more, how that will be useful to read this line. And this will go away very quickly with other outputs from sync flows. That is why I thought using a debug might be useful for us to see which items are been queued up.

Reply:
Maybe we can just mention we are queuing up necessary code syncs without listing them, since the following log flows will include each sync flow's execution, and move the list print out to debug

Reply:
That makes sense. And we can still keep list of sync flows that we are queuing up in a debug message.

Reply:
Updated ;)